src/router_core/exchange_bindings.c

```diff
@@ -0,0 +1,1336 @@
+        qdr_binding_t *binding = find_binding(core, identity, name);
+        if (!binding) {
+            query->status = QD_AMQP_NOT_FOUND;
+        } else {
+            qd_log(core->agent_log, QD_LOG_DEBUG,
+                   "Binding %s -> %s on exchange %s DELETED (id=%"PRIu64")",
+                   binding->key_str,
+                   binding->next_hop->next_hop_str,
+                   binding->exchange->name_str,
+                   binding->identity);
+            qdr_binding_free(binding);
+            query->status = QD_AMQP_NO_CONTENT;
+        }
+    }
+
+    qdr_agent_enqueue_response_CT(core, query);
+}
+
+
+// Binding GET
+//
+void qdra_config_binding_get_CT(qdr_core_t    *core,
+                                qd_iterator_t *name,
+                                qd_iterator_t *identity,
+                                qdr_query_t   *query,
+                                const char    *columns[])
+{
+    if (!identity && !name) {
+        query->status = QD_AMQP_BAD_REQUEST;
+        query->status.description = "No binding name or identity provided";
+        qd_log(core->agent_log, QD_LOG_ERROR, "Error performing READ of %s: %s",
+               config_binding_entity_type, query->status.description);
+    } else {
+        qdr_binding_t *binding = find_binding(core, identity, name);
+        if (binding == 0) {
+            query->status = QD_AMQP_NOT_FOUND;
+        }
+        else {
+            if (query->body) write_config_binding_map(binding, query->body);
+            query->status = QD_AMQP_OK;
+        }
+    }
+
+    qdr_agent_enqueue_response_CT(core, query);
+}
+
+
+// Binding GET first
+//
+void qdra_config_binding_get_first_CT(qdr_core_t *core, qdr_query_t *query, int offset)
+{
+    query->status = QD_AMQP_OK;
+
+    qdr_binding_t *binding = get_binding_at_index(core, offset);
+    if (!binding) {
+        query->more = false;
+        qdr_agent_enqueue_response_CT(core, query);
+        return;
+    }
+
+    if (query->body) write_config_binding_list(binding, query);
+    query->next_offset = offset + 1;
+    query->more = !!(DEQ_NEXT_N(exchange_list, binding) || DEQ_NEXT(binding->exchange));
+    qdr_agent_enqueue_response_CT(core, query);
+}
+
+
+// Binding GET-NEXT
+//
+void qdra_config_binding_get_next_CT(qdr_core_t *core, qdr_query_t *query)
+{
+    qdr_binding_t *binding = get_binding_at_index(core, query->next_offset);
+    if (binding) {
+        if (query->body) write_config_binding_list(binding, query);
+        query->next_offset++;
+        query->more = !!(DEQ_NEXT_N(exchange_list, binding) || DEQ_NEXT(binding->exchange));
+    } else
+        query->more = false;
+    qdr_agent_enqueue_response_CT(core, query);
+}
+
+
+// create a new iterator with its own copy of the data. caller must free both the
+// iterator and the returned data buffer.  Used for retaining data extracted
+// from query messages.
+static qd_iterator_t *clone_iterator(qd_iterator_t *iter, unsigned char **data)
```

Comment:
Can this function be included as a part of the iterator API instead of being a static function ? Like qd_clone_iterator() ?

Reply:
I don't see why not.   I'll update the patch.